src/hooks/xcm/useXcmBridge.ts

```diff
@@ -0,0 +1,402 @@
+  return Number(Number(amount).toFixed(decimals));
+};
+
+export function useXcmBridge(selectedToken?: Ref<ChainAsset>) {
+  const srcChains = ref<Chain[] | null>(null);
+  const destChains = ref<Chain[] | null>(null);
+
+  const srcChain = ref<Chain | null>(null);
+  const destChain = ref<Chain | null>(null);
+  const destParaId = ref<number>(parachainIds.SDN);
+  const tokens = ref<ChainAsset[] | null>(null);
+  const selectedTokenBalance = ref<string>('0');
+  const modal = ref<'src' | 'dest' | 'token' | null>(null);
+  const amount = ref<string | null>(null);
+  const errMsg = ref<string>('');
+  const isDisabledBridge = ref<boolean>(true);
+  const isNativeBridge = ref<boolean>(true);
+  const destEvmAddress = ref<string>('');
+
+  const store = useStore();
+  const router = useRouter();
+  const substrateAccounts = computed(() => store.getters['general/substrateAccounts']);
+  const { currentAccount } = useAccount();
+  const { xcmAssets } = useXcmAssets();
+  const currentNetworkIdx = computed(() => {
+    const chainInfo = store.getters['general/chainInfo'];
+    const chain = chainInfo ? chainInfo.chain : '';
+    return getProviderIndex(chain);
+  });
+  const { handleResult, handleTransactionError } = useCustomSignature({});
+  const { balance } = useBalance(currentAccount);
+  let relayChainApi: RelaychainApi | null = null;
+
+  const resetStates = (): void => {
+    isDisabledBridge.value = true;
+    amount.value = null;
+    errMsg.value = '';
+  };
+
+  const closeModal = (): boolean => modal.value === null;
+
+  const openModal = (scene: 'src' | 'dest' | 'token'): void => {
+    modal.value = scene;
+  };
+
+  const selectToken = (token: ChainAsset): void => {
+    store.commit('xcm/setSelectedToken', token);
+    modal.value = null;
+    resetStates();
+  };
+
+  const getSelectedTokenBal = async (): Promise<string> => {
+    if (!currentAccount.value || !srcChain.value || !relayChainApi) {
+      return '0';
+    }
+
+    const balance = await relayChainApi.getBalance(currentAccount.value);
+    return balance.toString();
+  };
+
+  const inputHandler = (event: any): void => {
+    amount.value = event.target.value;
+    // check if recipient account has non-zero native asset. (it cannot be transferred to an account with 0 nonce)
+    isDisabledBridge.value =
+      !amount.value ||
+      Number(amount.value) === 0 ||
+      Number(amount.value) > Number(formattedRelayChainBalance.value) ||
+      balance.value.lten(0);
+    errMsg.value = '';
+  };
+
+  const selectChain = (chainId: number): void => {
+    const isSrcChain = modal.value === 'src';
+    const chain = CHAINS.find((it) => it.id === chainId);
+    if (!chain) return;
+
+    if (isSrcChain) {
+      srcChain.value = chain;
+    } else {
+      destChain.value = chain;
+    }
+    modal.value = null;
+    resetStates();
+  };
+
+  const chainIcon = computed<{ src: string; dest: string }>(() => {
```

Comment:
I don't think this is a scalable solution. Why don't we create an interface that handles the chain data like name and icon (and maybe the asset info too), and move that to a config file?

For example,
```
interface XcmNetwork {
   name: 'Polkadot';
   icon: './path/to/dot.png';
   endpoint: 'wss://...';
   asset: XcmAsset;
}
```

and create an object with the type:
```
interface XcmChannel {
   astar: XcmNetwork[];
   shiden: XcmNetwork[];
   shibuya: XcmNetwork[];
}
```

This is just an example, but because we plan to add more channels and we want this process to be scalable, code structure is key I think